.changeset/nervous-scissors-roll.md

```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+Fix automatic detection of Worker format when using `typeof module`
```

Comment:
Changesets should follow the guidelines set out in the CONTRIBUTORS doc, and contain useful information to changelog readers. There is loads in the PR description for this. So let's include that here too.


```suggestion
fix: correctly detect "Service Worker" format even if using `typeof module`

Wrangler automatically detects whether your code is a modules or service-worker format Worker based on the presence of a default export. This check currently works by building your entrypoint with esbuild and looking at the output metafile.

Previously, we were passing format: "esm" to esbuild when performing this check, which enables format conversion. This may introduce export default into the built code, even if it wasn't there to start with, resulting in incorrect format detections.

This change removes format: "esm" which disables format conversion when bundling is disabled: https://esbuild.github.io/api/#format.
```

Reply:
By the way, this format is only a convention. We can change it if it is not serving its purpose.